mmengine/runner/loops.py

```diff
@@ -432,8 +445,10 @@ def run(self) -> dict:
         self.runner.call_hook('before_test_epoch')
         self.runner.model.eval()
         for idx, data_batch in enumerate(self.dataloader):
+            if self.num_batch_per_epoch is not None:
+                if idx > self.num_batch_per_epoch:
+                    break
             self.run_iter(idx, data_batch)
-
         # compute metrics
         metrics = self.evaluator.evaluate(len(self.dataloader.dataset))
```

Comment:
We also need to re-calculate the real length of dataset when self.num_batch_per_epoch is not None

Reply:
Same as ValLoop